homeassistant/components/battery.py

```diff
@@ -0,0 +1,180 @@
+from homeassistant.helpers.entity import Entity
+from homeassistant.const import ATTR_BATTERY_LEVEL
+
+DEPENDENCIES = ['device_tracker']
+
+# domain for the component
+DOMAIN = 'battery'
+
+# entity attributes
+ATTR_STATUS = 'status'
+ATTR_BATTERY_TYPE = 'battery type'
+ATTR_ATTRIBUTE = 'attribute'
+
+# default type
+DEFAULT_TYPE = 'attribute'
+
+# default attribute
+DEFAULT_ATTRIBUTE = ATTR_BATTERY_LEVEL
+
+# default unit_of_measurement
+DEFAULT_UNIT = '%'
+
+# default charged_level
+DEFAULT_CHARGED = 100
+
+# Shortcut for the logger
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup(hass, config):
+    """ get the battery details from configuration.yaml"""
+
+    batteries = []
+    if config.get(DOMAIN) is None:
+        return False
+
+    for bat, bat_config in config[DOMAIN].items():
+
+        if not isinstance(bat_config, dict):
+            _LOGGER.error("Missing configuration data for battery %s", bat)
+            continue
+
+        battery_type = bat_config.get('type', DEFAULT_TYPE)
+        attribute = bat_config.get('attribute', DEFAULT_ATTRIBUTE)
+        device_entity_id = bat_config.get('entity_id')
+        unit_of_measurement = bat_config.get('unit_of_measurement',
+                                             DEFAULT_UNIT)
+        charged_level = bat_config.get('charged_level', DEFAULT_CHARGED)
+
+        entity_id = DOMAIN + '.' + bat
+
+        battery = Battery(hass, battery_type, attribute, device_entity_id,
+                          unit_of_measurement, charged_level)
+        battery.entity_id = entity_id
+
+        battery.update_ha_state()
+        battery.check_battery_state_change(None, None, None)
+
+        batteries.append(battery)
+        # main command to monitor battery of device
+        track_state_change(hass, battery.device,
+                           battery.check_battery_state_change)
+
+    if not batteries:
+        _LOGGER.error("No batteries added")
+        return False
+
+    # Tells the bootstrapper that the component was successfully initialized
+    return True
+
+
+class Battery(Entity):  # pylint: disable=too-many-instance-attributes
+    """ Represents a Battery in Home Assistant. """
+    def __init__(self, hass, battery_type, attribute, device_entity_id,
+                 unit_of_measurement, charged_level):
+        # pylint: disable=too-many-arguments
+        self.hass = hass
+        self.device = device_entity_id
+        self._battery_type = battery_type
+        self._attribute = attribute
+        self._batterylevel = 'not set'
+        self._status = 'not set'
+        self._unit_of_measurement = unit_of_measurement
+        self._charged_level = charged_level
+
+    @property
+    def state(self):
+        return self._batterylevel
+
+    @property
+    def unit_of_measurement(self):
+        """ Unit of measurement of this entity """
+        return self._unit_of_measurement
+
+    @property
+    def state_attributes(self):
```

Comment:
You should use device_state:attributes instead. See https://github.com/balloob/home-assistant/pull/1157.